Sources/NIOExtras/LengthFieldPrepender.swift

```diff
@@ -0,0 +1,132 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+public enum LengthFieldPrependerError: Error {
+    case messageDataTooLongForLengthField
+}
+
+///
+/// An encoder that takes a `ByteBuffer` message and prepends the number of bytes in the message.
+/// The length field is always the same fixed length specified on construction.
+/// These bytes contain a binary specification of the message size.
+///
+/// For example, if you received a packet with the 3 byte length (BCD)...
+/// Given that the specified header length is 1 byte, there would be a single byte prepended which contains the number 3
+///     +---+-----+
+///     | A | BCD | ('A' contains 0x0003)
```

Comment:
Nit: should be `0x03`, `0x0003` is two bytes.

Reply:
Haha, true, the perils of changing an example at the last minute. Thanks.